addons/website/static/src/js/content/website_root.js

```diff
@@ -254,6 +254,29 @@ var WebsiteRoot = publicRootData.PublicRoot.extend(KeyboardNavigationMixin, {
         var res = this._unslugHtmlDataObject('seo-object');
         ev.data.callback(res);
     },
+    /**
+     * Gives the given button a loading effect by disabling it and adding a `fa`
+     * spinner icon.
+     * The existing button `fa` icons will be hidden through css.
+     *
+     * @private
+     * @param {OdooEvent} ev
+     * @param {object} ev.data
+     * @param {HTMLElement} ev.data.btn - the button to disable/load
+     * @param {Boolean} [ev.data.spin=false] - true to disable, false to enable
+     */
+    _onBtnSpinRequest: function (ev) {
+        // TODO: Should we prevent this in edit mode? and allow for "Save" with
+        //       a force parameter such as ev.data.forceInEditMode
+        const $btn = $(ev.data.btn);
+        $btn.toggleClass('o_website_btn_loading disabled', ev.data.spin || false);
+        $btn.prop('disabled', ev.data.spin || false);
+        if (ev.data.spin) {
+            $btn.prepend('<span class="fa fa-refresh fa-spin"/><span class="o_website_btn_loading_space mr-2"/>');
+        } else {
+            $btn.find('.fa-spin, .o_website_btn_loading_space').remove();
+        }
+    },
```

Comment:
Using the trigger_up system for this is not a good idea. trigger_up is only useful if you need info only an ancestor has, which is not the case at all here. This better be a standalone function near the definition of `makeButtonHandler`. Which you could use like this:
```javascript
const restore = addButtonLoadingEffect(myButton);
// ... async stuff ...
restore();
```